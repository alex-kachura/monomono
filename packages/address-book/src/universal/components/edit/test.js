import React from 'react';
import { render } from 'react-testing-library';
import { renderProviders } from '@oneaccount/react-foundations/lib/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { DefaultThemeProvider } from '@beans/theme';
import EditPage from './';

describe('[Component: EditPage]', () => {
  let component;

  beforeEach(() => {
    const providers = renderProviders({
      appConfig: {
        csrf: 'mock-csrf',
        getLocalePhrase: (key) => key,
        rootPath: '/mock/math',
      },
      children: (
        <DefaultThemeProvider>
          <BrowserRouter>
            <EditPage />
          </BrowserRouter>
        </DefaultThemeProvider>
      ),
    });

    component = render(providers);
  });

  it('should render correctly', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
