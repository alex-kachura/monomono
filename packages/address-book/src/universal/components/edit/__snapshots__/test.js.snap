// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: EditPage] should render correctly 1`] = `
<DocumentFragment>
  <h1
    class="base-components__BaseElement-sc-150tvch-0 typography___styledElement-sc-186l0ce-0 bISoeZ"
  >
    pages.edit.title
  </h1>
  <div
    class="sc-bdVaJa iWvfag"
  >
    <form
      autocomplete="off"
      method="POST"
      novalidate=""
    >
      <div
        class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
      >
        <div
          class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="name"
          >
            Name 
          </label>
        </div>
        <div
          class="beans-form-group__children"
        >
          <input
            class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
            id="name"
            name="name"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
      >
        <div
          class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="age"
          >
            Age 
          </label>
        </div>
        <div
          class="beans-form-group__children"
        >
          <input
            class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
            id="age"
            name="age"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
      >
        <div
          class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="password"
          >
            Password 
          </label>
        </div>
        <div
          class="beans-form-group__children"
        >
          <input
            class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
            id="password"
            name="password"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
      >
        <div
          class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="confirmPassword"
          >
            Confirm Password 
          </label>
        </div>
        <div
          class="beans-form-group__children"
        >
          <input
            class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
            id="confirmPassword"
            name="confirmPassword"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
      >
        <div
          class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
        >
          <label
            class="styled__StyledLabel-sc-28yvb4-0 bPkari"
            for="postcode"
          >
            Postal Code 
          </label>
        </div>
        <div
          class="beans-form-group__children"
        >
          <input
            class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
            id="postcode"
            name="postcode"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        aria-disabled="false"
        class="styled__BaseButton-sc-3mnirm-0 find-address is-find-address sc-htpNat jqgOZr styled__PrimaryButton-sc-3mnirm-2 evotEg"
        role="button"
        type="button"
      >
        <span
          class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 lnNmDn"
        >
          <span
            class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__PrimarySpan-sc-3mnirm-5 hrUcvf"
          >
            Find Address
          </span>
        </span>
      </button>
      <div
        class="find-address is-find-address"
      >
        <br />
         
        <br />
         
        <br />
        <br />
        <a>
          Enter Address Manually
        </a>
        <br />
      </div>
      <div
        class="display-address is-find-address"
      >
        <div
          class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
        >
          <div
            class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
          >
            <label
              class="styled__StyledLabel-sc-28yvb4-0 bPkari"
              for="address-line1"
            >
              Address Line 1 
            </label>
          </div>
          <div
            class="beans-form-group__children"
          >
            <input
              class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
              id="address-line1"
              name="address-line1"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
        >
          <div
            class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
          >
            <label
              class="styled__StyledLabel-sc-28yvb4-0 bPkari"
              for="address-line2"
            >
              Address Line 2 
              <span
                class="base-components__BaseElement-sc-150tvch-0 styled__OptionalSpan-wt4ift-0 jPEuop"
              >
                (optional)
              </span>
            </label>
          </div>
          <div
            class="beans-form-group__children"
          >
            <input
              class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
              id="address-line2"
              name="address-line2"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
        >
          <div
            class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
          >
            <label
              class="styled__StyledLabel-sc-28yvb4-0 bPkari"
              for="address-line3"
            >
              Address Line 3 
              <span
                class="base-components__BaseElement-sc-150tvch-0 styled__OptionalSpan-wt4ift-0 jPEuop"
              >
                (optional)
              </span>
            </label>
          </div>
          <div
            class="beans-form-group__children"
          >
            <input
              class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
              id="address-line3"
              name="address-line3"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="base-components__RootElement-sc-150tvch-1 sc-bwzfXH gYKmoc styled__FormGroupContainer-wt4ift-1 hVjJdM"
        >
          <div
            class="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
          >
            <label
              class="styled__StyledLabel-sc-28yvb4-0 bPkari"
              for="town"
            >
              town 
            </label>
          </div>
          <div
            class="beans-form-group__children"
          >
            <input
              class="base-components__BaseInput-sc-150tvch-2 styled-ep7354-0 eyeGHp"
              id="town"
              name="town"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <input
        name="_csrf"
        type="hidden"
        value="mock-csrf"
      />
      <button
        aria-disabled="false"
        class="styled__BaseButton-sc-3mnirm-0 sc-ifAKCX hhWYMO styled__PrimaryButton-sc-3mnirm-2 evotEg"
        role="button"
        type="submit"
      >
        <span
          class="base-components__BaseElement-sc-150tvch-0 styled__InnerContainer-sc-3mnirm-1 lnNmDn"
        >
          <span
            class="base-components__BaseElement-sc-150tvch-0 styled__BaseSpan-sc-3mnirm-4 beans-button__text styled__PrimarySpan-sc-3mnirm-5 hrUcvf"
          >
            pages.edit.submit-btn
          </span>
        </span>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
